### PR TITLE
Add Flatness boundary conditions for XCTS

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Boundary conditions representing flat spacetime.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

**Comment by nilsvu** [comment]
Rebased since this has been open for a while.

**Review by kidder — DISMISSED** [review]

File: tests/Unit/Elliptic/Systems/Xcts/BoundaryConditions/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "${LIBRARY_SOURCES}"
+  ""
+  )
+
+target_link_libraries(
+  ${LIBRARY}
+  PRIVATE
+  DataStructures
+  DomainStructure
+  XctsBoundaryConditions
```
Comment (moxcodes) [inline tests/Unit/Elliptic/Systems/Xcts/BoundaryConditions/CMakeLists.txt]: alphabet

File: src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Xcts/BoundaryConditions/Flatness.hpp"
+
+#include <cstddef>
```
Comment (moxcodes) [inline src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.cpp:7]: `#include <algorithm>` (for std::fill)

File: src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.hpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+#include <string>
```
Comment (moxcodes) [inline src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.hpp:9]: `#include <memory>`

File: src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.hpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+struct FlatnessImpl {
+  using options = tmpl::list<>;
+  static constexpr Options::String help =
+      "Impose flat spacetime at this boundary.";
+
+  using argument_tags = tmpl::list<>;
+  using volume_tags = tmpl::list<>;
+
+  static void apply(gsl::not_null<Scalar<DataVector>*> conformal_factor,
```
Comment (moxcodes) [inline src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.hpp:33]: Sorry, I haven't been following the other features of the Xcts system -- can you explain this interface? I would have anticipated the arguments to have something to do with the tag lists, but perhaps these are the 'standard' system variables for Xcts?

**Review by moxcodes — COMMENTED** [review]
Feel free to squash any changes -- I only have nitpick comments

File: src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.hpp (hunk 4 above)
Comment (nilsvu) [reply src/Elliptic/Systems/Xcts/BoundaryConditions/Flatness.hpp:33]: You should find all answers in the docs of the base class (I hope): https://spectre-code.org/classelliptic_1_1BoundaryConditions_1_1BoundaryCondition.html. Indeed the conformal factor, the lapse-times-conformal-factor and the shift-excess are the XCTS system variables (#2667). The XCTS system also supports solving only for subsets of the variables. Let me know if the docs leave anything unclear :)

**Comment by nilsvu** [comment]
Rebased and squashed in the missing includes. Thanks for reviewing @kidder and @moxcodes!

**Review by moxcodes — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**